dev/com.ibm.websphere.appserver.features/test/src/com/ibm/ws/feature/tests/VisibilityTest.java

```diff
@@ -273,7 +273,12 @@ public void testVisibility2() {
             File featureFile = featureInfo.getFeatureFile();
             String fileName = featureFile.getAbsolutePath();
             fileName = fileName.replace('\\', '/');
-            if (!fileName.contains("/visibility/" + visibility + "/")) {
+            String expectedPathName = "/visibility/" + visibility + "/";
+            if ("public".equals(visibility)) {
+                expectedPathName += (featureInfo.getShortName() + '/');
+            }
+            expectedPathName += (featureName + ".feature");
+            if (!fileName.endsWith(expectedPathName)) {
                 errorMessage.append("Found issues with " + featureName + '\n');
                 errorMessage.append("     Feature visibility " + visibility + " doesn't match the directory the feature is in.\n");
```

Comment:
Seems this message could be improved to indicate more than just the "visibility" may have a problem... `Feature visibility public doesn't match the directory the feature is in.` could be confusing, as it may be in the `public` directory.  Perhaps just change the `visibility` variable to `expectedPathName`?